clients/roscpp/src/libros/connection.cpp

```diff
@@ -93,7 +93,9 @@ boost::signals2::connection Connection::addDropListener(const DropFunc& slot)
 void Connection::removeDropListener(const boost::signals2::connection& c)
 {
   boost::recursive_mutex::scoped_lock lock(drop_mutex_);
-  c.disconnect();
+  if (c.connected()) {
```

Comment:
This check doesn't seem to be necessary? The `disconnect` method already checks for this.

Reply:
Yes. I will update codes.